src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -169,10 +171,37 @@ public static int Size
         [NonVersionable]
         public unsafe void* ToPointer() => _value;
 
-        public override unsafe string ToString() =>
-            ((nint)_value).ToString(CultureInfo.InvariantCulture);
+        public static IntPtr MaxValue => (IntPtr)nint.MaxValue;
+        public static IntPtr MinValue => (IntPtr)nint.MinValue;
 
-        public unsafe string ToString(string format) =>
-            ((nint)_value).ToString(format, CultureInfo.InvariantCulture);
+        public int CompareTo(object? value) => ((nint)this).CompareTo(value);
```

Comment:
nit: I think the conventionis to do `((nint)_value)` rather than `((nint)this)`

Reply:
Do both forms generate same IL? We should make sure that the native code generated for these is good.

Reply:
@john-h-k, could you get an assembly diff for the two forms?

I would hope the JIT folds them the same, but the latter ends up going through the implicit conversion API (vs directly converting the field).

We were already using the former for existing APIs (like `ToString()`).